ClinGen/clincoded
# Connects to #1267. React 15 Upgrade.

For a list of changes made during this upgrade, please refer to https://github.com/ClinGen/clincoded/issues/1267.

**Technical notes:**
1. Due to React 15 warning the use of non-standard prop/attribute in DOM element associated with modals, the use of `ModalMixin` had been deprecated and a rewritten `ModalComponent` had been implemented.
2. In addition to upgrading the ReactJS library from 0.14 to 15.4.2, the **npm** package, `moment.js`, had also been upgraded from 2.17 to 2.18 in the process of addressing failing BDD test on Travis CI.

**Known issues:**
1. In this PR, the following "Mixins" have not been deprecated: `FormMixin`, `RestMixin`, `CurationMixin`, `CuratorHistory`.
2. Console warning on development build of React when it is not a production instance (for reference: https://facebook.github.io/react/docs/optimizing-performance.html#use-the-production-build).
![image](https://cloud.githubusercontent.com/assets/6956310/24575306/c635f2fc-1656-11e7-9dc5-d52c144c210f.png)

**Steps to test:**
_While this upgrade virtually affects all parts of the interfaces, the following few areas particularly need a bit more attention:_
1. Test all curation form pages (i.e. group, family, individual, and experimental) that have text inputs and dropdown selections in the GCI. Expect to see _no_ console errors/warnings associated with the respective events.
2. Test all tabs (and the summary page) that have text inputs and dropdown selections in the VCI. Expect to see _no_ console errors/warnings associated with the respective events.
3. Test pop-up modals (e.g. "Create GDM", "Add/Edit OMIM", "Add PMID", etc.) in both of the GCI and VCI. Expect to see the same behaviors as seen in production, and _no_ console errors/warnings associated with the respective events.
4. On the family curation form page, verify text inputs that only accept integers (e.g. "Tested Individuals" section in the "Family -- Segregation" panel). Expect to see error message being displayed below those text inputs upon form submission if the values are not integers.
5. On the GDM list (`^/gdm`) and all Interpretation list (`^/interpretation`) pages, expect to see the date/time displayed in the tables to match those in production.
6. On the GDM list (`^/gdm`) and all Interpretation list (`^/interpretation`) pages, enter a gene name or disease name in the text input field and expect to see the list to be properly filtered as seen in production.
